protocol/lib/bytes.go

```diff
@@ -5,45 +5,38 @@ import (
 	"strconv"
 )
 
-func Uint32ToBytes(i uint32) []byte {
-	bytes := make([]byte, 4)
-	binary.LittleEndian.PutUint32(bytes, i)
-	return bytes
-}
-
-func Int32ToBytes(i int32) []byte {
+// Bit32ToBytes converts 32-bit types into a 4-byte slice in little-endian format.
+func Bit32ToBytes[T uint32 | int32](i T) []byte {
```

Comment:
I did like the original name format like `Int32ToBytes` since its clear that the conversion is for an integer, `Bit32` is a little to vague and could represent things like `Float` and made it harder when reading the code where it was used to understand what was happening.